lib/portage/eapi.py

```diff
@@ -2,33 +2,54 @@
 # Distributed under the terms of the GNU General Public License v2
 
 import collections
-import operator
-import types
-
-from portage import eapi_is_supported
+from functools import lru_cache
 
 
+@lru_cache(None)
 def eapi_has_iuse_defaults(eapi):
+    if eapi is None:
+        return True
+
     return eapi != "0"
 
 
+@lru_cache(None)
 def eapi_has_iuse_effective(eapi):
+    if eapi is None:
+        return False
+
     return eapi not in ("0", "1", "2", "3", "4", "4-python", "4-slot-abi")
```

Comment:
```suggestion
    return eapi not in (None, "0", "1", "2", "3", "4", "4-python", "4-slot-abi")
```

and so on…

Reply:
resolving this comment, leaving the other one open.